From e48ed8d370432f40bb1c0e86c6976a82a5686d60 Mon Sep 17 00:00:00 2001
From: Michael Heimpold <mhei@heimpold.de>
Date: Sat, 17 Feb 2024 22:42:45 +0100
Subject: [PATCH] Set library properties to fix Yocto's QA error

Without this, Yocto barks:

QA Issue: -dev package rpclib-dev contains non-symlink .so '/usr/lib/librpc.so' [dev-elf]

Signed-off-by: Michael Heimpold <mhei@heimpold.de>
---
 CMakeLists.txt | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 721b09312a53..e55063102e54 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,6 +106,11 @@ add_library(${PROJECT_NAME}
   ${DEP_HEADERS}
   ${RPCLIB_HEADERS})
 
+set_target_properties(${PROJECT_NAME}
+  PROPERTIES
+    VERSION ${PROJECT_VERSION}
+    SOVERSION 1)
+
 # Perform steps and checks required for MSVC support
 rpclib_msvc_support()
 
-- 
2.34.1

